libs/utils/env.py

```diff
@@ -603,8 +604,13 @@ def _init_platform_smp(self):
             # Try loading frequencies using the cpufreq module
             for cluster_id in self.platform['clusters']:
                 core_id = self.platform['clusters'][cluster_id][0]
-                self.platform['freqs'][cluster_id] = \
-                    self.target.cpufreq.list_frequencies(core_id)
+                try:
+                    self.platform['freqs'][cluster_id] = \
+                        self.target.cpufreq.list_frequencies(core_id)
+                except TargetError:
```

Comment:
> You mean that if specify "cpufreq" among the modules to be loaded, than we should raise in case one of these modules is not properly working?
> [...] Consider that if the cpufreq module has been loaded, than we should get a KeyError exception.
While if cpufreq.list_frequencies() should fails that we can quite likely assume that OPPs are not expose... the only other possible issue is a communication error..
With the current devlib implementation and a good kernel yes, but any number of things could change or break in the kernel or devlib so that we can't read scaling_available_frequencies, and we should be able to distinguish those failures (and a communication error) from the frequencies just being unavailable.

> Regarding the ABI checks I would not assume that x86 never reports frequencies, is that the case?
Good point, I don't know.

> Regarding scaling_driver, where do we get this info? You mean target.cupfreq.get_all_governors()?
Nope, `scaling_driver` is a separate value:

```
brendan@debian-arm64-jessie:~$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_driver 
arm-big-little

brendan-thinkstation:~$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor 
powersave
```

Devlib doesn't expose it at the moment but we could easily add it.